subsys/net/lib/fota_download/src/fota_download.c

```diff
@@ -431,6 +431,7 @@ int fota_download(const char *host, const char *file,
 	struct download_client_cfg config = {
 		.pdn_id = pdn_id,
 		.frag_size_override = fragment_size,
+		.family = family
```

Comment:
suggest adding the following to `fota_download`'s Kconfig:

```
choice
	prompt "Address family to be used for the FOTA download"
	default FOTA_DOWNLOAD_ADDR_FAM_UNSPEC

config FOTA_DOWNLOAD_ADDR_FAM_IPV4
	bool "IPv4 (AF_INET)"
config FOTA_DOWNLOAD_ADDR_FAM_IPV6
	bool "IPv6 (AF_INET6)"
config FOTA_DOWNLOAD_ADDR_FAM_UNSPEC
	bool "Unspecified (AF_UNSPEC)"

endchoice
```

and just set the family here:

```
	if (IS_ENABLED(CONFIG_FOTA_DOWNLOAD_ADDR_FAM_IPV4)) {
		config.family = AF_INET;
	} else if (IS_ENABLED(CONFIG_FOTA_DOWNLOAD_ADDR_FAM_IPV6)) {
		config.family = AF_INET6;
	} else {
		config.family = AF_UNSPEC;
	}
```

Reply:
I will go ahead and make these changes. Better a Kconfig than no control.